ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaCanBeSilmpifiedAnalyzer.fs

```diff
@@ -0,0 +1,51 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Analyzers
+
+open JetBrains.Util
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings.Errors
+
+[<ElementProblemAnalyzer(typeof<ILambdaExpr>, HighlightingTypes = [|typeof<LambdaCanBeSimplifiedWarning>|])>]
+type LambdaCanBeSimplifiedAnalyzer() =
+    inherit ElementProblemAnalyzer<ILambdaExpr>()
+
+    let rec compareArg (pat: IFSharpPattern) (arg: IFSharpExpression) =
+        match pat.IgnoreInnerParens(), arg.IgnoreInnerParens() with
+        | :? ITuplePat as pat, (:? ITupleExpr as tuple) -> compareArgsSeq pat.PatternsEnumerable tuple.ExpressionsEnumerable
```

Comment:
Naming: why the tuple pattern is called `pat` and the tuple expression is called `tuple`? They both represent tuples anyway. `tuplePat`+`tupleExpr` or `pat`+`expr`?